File: ICP.SQLite/Models/Contractor.cs
using System.ComponentModel.DataAnnotations;

namespace ICP.SQLite.Models
{
    public class Contractor
    {
        [Key]
        [Required]
        public int Id { get; set; }

        [StringLength(50)]
        public string Name { get; set; }

        //public Address Address { get; set; }

        [StringLength(20)]
        public string Phone { get; set; }

        [StringLength(10)]
        public string Type { get; set; }

        [StringLength(6)]
        public string HealthStatus { get; set; }

    }
}

File: ICP.Web/ClientApp/src/app/Models/contractor.ts
export class Contractor {
  constructor(
    public id: number,
    public name: string) {
  }
}

export class ContractorDetail extends Contractor {
  constructor(
    public id: number,
    public name: string,
    public phone: string) {
      super(id, name);
  }
}
